torchrl/data/tensordict/tensordict.py

```diff
@@ -1159,24 +1159,24 @@ def unsqueeze(self, dim: int) -> TensorDictBase:
             inv_op_kwargs={"dim": dim},
         )
 
-    def squeeze(self, dim: int) -> TensorDictBase:
+    def squeeze(self, dim: Optional[int]) -> TensorDictBase:
         """Squeezes all tensors for a dimension comprised in between `-td.batch_dims+1` and `td.batch_dims-1` and returns them in a new tensordict.
 
         Args:
             dim (int): dimension along which to squeeze
 
         """
-        if dim < 0:
+        if dim and dim < 0:
```

Comment:
I guess you'd like to check that `dim` is not `None`
I'd rather write `if dim is not None and dim < 0` instead for clarity.